lib/AsyncUtils.lua

```
@@ -0,0 +1,131 @@
+local TableUtils = require(script.Parent.TableUtils)
+local Promise = require(script.Parent.Parent.Promise)
+local AsyncUtils = {}
+
+local baseRandomStream = Random.new()
+
+function AsyncUtils.parallel(things)
```

how does this work wrt error handling?

---

falls back to usual Promise.all behaviour which is fail fast and once if any child fails